cloudflare/resource_cloudflare_filter.go

```diff
@@ -40,6 +40,10 @@ func resourceCloudflareFilter() *schema.Resource {
 			"expression": {
 				Type:     schema.TypeString,
 				Required: true,
+				// Suppress the diff shown if there's trailing carriage returns at the end of the filter (needed for heredocs)
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return strings.TrimRight(new, "\n") == old
```

Comment:
Have we considered other whitespace characters and combination? For example, does multiple spaces result in the same behaviour? What about the use case where Windows users might introduce a carriage return and line feed (`\r\n`) into the expression?

Reply:
You're right, I played a bit with the Cloudflare API it always strips tabs, whitespaces and carriage-returns at the beginning **and** the end of expressions and that's also valid for different combinations, so I adapted my code accordingly.